ruby/src/IceRuby/Operation.cpp

```diff
@@ -571,8 +571,14 @@ IceRuby::OperationI::unmarshalException(const vector<Ice::Byte>& bytes, const Ic
 
     try
     {
-        Ice::UserExceptionFactoryPtr factory = new UserExceptionFactory;
-        is.throwException(factory);
+        is.throwException([](const auto& id)
+            {
+                ExceptionInfoPtr info = lookupExceptionInfo(id);
+                if(info)
```

Comment:
What happens if the lookup doesn't find the ExceptionInfoPtr?

Looks like we don't do anything, which does not sound correct.

Reply:
if the factory doesn't throw, we Slice the exception and try again with the sliced format, or throw UnknownUserException for the last slice.

Reply:
Would be nice to add a comment.